net: tcp: Recalculate the tcp checksum when changing tcp header.

junelizh opened the pull request:
Tcp will retransmit the lost packet, when received one tcp data packet
befor retransmission, the tcp acknowledgment number will be changed and
set to the retransmitted packet, need calculate checksum again.

Signed-off-by: june li <junelizh@foxmail.com>

jukkar commented on subsys/net/ip/tcp.c:
```diff
@@ -696,18 +696,29 @@ int net_tcp_send_pkt(struct net_pkt *pkt)
 	}
 
+	if (calc_chksum) {
+		NET_TCP_HDR(pkt)->chksum = 0;
```
The NET_TCP_HDR() macro was removed by #691 so this part needs changing.

You can call

net_tcp_set_chksum(pkt, pkt->frags);

to calculate the checksum.

jukkar reviewed (changes requested):
Looks ok, the commit message has typos and is hard to understand. Please change it to something like this:

If the TCP data packet needs to be re-sent after the packet is lost, then the acknowledgment number will be changed. This then means that the the TCP checksum needs to be recalculated too.

junelizh committed:
net: tcp: Recalculate the tcp checksum when changing tcp header.

If the TCP data packet needs to be re-sent after the packet is lost,
then the acknowledgment number will be changed. This then means that
the TCP checksum needs to be recalculated too.

Signed-off-by: june li <junelizh@foxmail.com>

junelizh commented:
Have modify the code and update commit message,
thanks for the commit message.
